src/tespy/tools/analyses.py

```diff
@@ -0,0 +1,759 @@
+                del group_data[fkt_group]
+                # recursive call in case multiple components are attached in
+                # a line without any conversion
+                self.remove_transit_groups(group_data)
+
+    def generate_plotly_sankey_input(
+            self, node_order=[], colors={}, display_thresold=1e-3):
+        """Generate input data for sankey plots.
+
+        Only exergy flow above the display threshold is included. All
+        component groups with transit only (one input and one output) are cut
+        out of the display.
+
+        Parameters
+        ----------
+        node_order : list
+            Order for the nodes in the sankey diagram (optional).
+            In case no order is passed, a generic order will be used.
+
+        colors : dict
+            Dictionary containing a color for every stream type (optional).
+            Stream type is the key, the color is the corresponding value.
+            Stream types are:
+
+            - :code:`E_F`, :code:`E_P`, :code:`E_L`, :code:`E_D`
+            - names of the pure fluids of the tespy Network
+            - :code:`mix` (used in case of any gas mixture)
+            - labels of internal busses
+
+            In case no colors are passed, the matplotlib :code:`Set1` colormap
+            will be applied.
+
+        Returns
+        -------
+        tuple
+            Tuple containing the links and node_order for the plotly sankey
+            diagram.
+        """
+        group_data = self.group_data.copy()
+
+        for fkt_group, data in self.group_data.items():
+            group_data[fkt_group] = group_data[fkt_group][
+                data['value'].abs() >= display_thresold]
+
+        self.remove_transit_groups(group_data)
+
+        if len(node_order) == 0:
+            node_order = (
+                ['E_F'] + [b.label for b in self.E_F] +
+                [fkt_group for fkt_group in self.group_overview.index] +
+                [b.label for b in self.internal_busses + self.E_P + self.E_L] +
+                ['E_P', 'E_L', 'E_D'])
+
+        if len(colors) == 0:
+            cmap = cm.get_cmap('Set1')(np.linspace(0.0, 1.0, 10))
+            cmap[:, 0:3] *= 255
+            cmap[:, 3] *= 0.75
+            rgba_list = []
+            for i in range(cmap.shape[0]):
+                rgba_list += [
+                    'rgba(' +
+                    str(cmap[i, 0]) + ', ' +
+                    str(cmap[i, 1]) + ', ' +
+                    str(cmap[i, 2]) + ', ' +
+                    str(cmap[i, 3]) + ')']
+
+            colors = {}
+            colors['E_P'] = rgba_list[0]
+            colors['E_F'] = rgba_list[1]
+            colors['E_D'] = rgba_list[2]
+            colors['E_L'] = rgba_list[3]
+            i = 4
+            for f in self.nw.fluids:
+                colors[f] = rgba_list[i]
+                i += 1
+            for b in self.internal_busses:
+                colors[b.label] = rgba_list[i]
+                i += 1
```

Comment:
Generate the dictionary, and update with the passed dictionary afterwards. By this, the user does not need to specify a full dictionary, but only the colors he wants to change.